operator/pkg/object/objects_test.go

```diff
@@ -587,3 +587,126 @@ func TestK8sObject_Equal(t *testing.T) {
 		})
 	}
 }
+
+func TestK8sObject_ResolveConflict(t *testing.T) {
+	getK8sObject := func(ystr string) *K8sObject {
+		o, err := ParseYAMLToK8sObject([]byte(ystr))
+		if err != nil {
+			return nil
+		}
+		// Ensure that json data is in sync.
+		o.json, _ = o.JSON()
```

Comment:
it seems like we should maybe be taking care of this inside the function. i don't recall why it wasn't but since we're having to do this workaround maybe it should be fixed at the source.

Reply:
@ostromart The original function will return the json string if there is one already exists. In our case, since we are make changes to the object by accessing the map (which hold the actual data), we need to really make sure that the json and yaml bytes are truly consistent with the actual object.

Reply:
let's create o.Update() function or something like that to make this a bit clearer